src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala

```diff
@@ -42,8 +42,17 @@ trait KorbHandler extends KorbStatusHandler
   def upsertKorb(lieferung: Lieferung, abo: Abo, abotyp: IAbotyp)(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): (Option[Korb], Option[Korb]) = {
     stammdatenWriteRepository.getKorb(lieferung.id, abo.id) match {
       case None if (lieferung.lieferplanungId.isDefined) =>
-        val abwCount = stammdatenWriteRepository.countAbwesend(lieferung.id, abo.id)
-        val status = calculateKorbStatus(abwCount, abo.guthaben, abotyp.guthabenMindestbestand)
+        val status = abo match {
+          case zusatzAbo: ZusatzAbo =>
+            val mainAbo = stammdatenWriteRepository.getById(depotlieferungAboMapping, zusatzAbo.hauptAboId) orElse
```

Comment:
In order to do things always the same way I suggest you create a function StammdatenReadRepositoryAsync.getHauptabo. See StammdatenReadRepositoryAsyncImpl.getAboDetail(id: AboId)

Reply:
done